Nightly test failed notification not working

The nightly test failed (https://github.com/aiidateam/aiida-core/actions/runs/3129772900) but the notification not send with this error.

```
Run rtCamp/action-slack-notify@v2
  env:
    SINGULARITY_ROOT: /opt/hostedtoolcache/singularity/3.8.7/x64
    pythonLocation: /opt/hostedtoolcache/Python/3.10.7/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.7/x64/lib
    SLACK_WEBHOOK: 
    SLACK_ICON: https://www.materialscloud.org/discover/images/0ba0a17d.aiida-logo-1[2](https://github.com/aiidateam/aiida-core/actions/runs/3129772900/jobs/5079265531#step:12:2)8.png
    SLACK_CHANNEL: dev-aiida-core
    SLACK_COLOR: b60205
    SLACK_TITLE: Nightly build of `aiida-core/main` failed
    SLACK_MESSAGE: The tests of the `nightly.yml` GHA worklow failed.
/usr/bin/docker run --name ghcriortcampactionslacknotifyv220_da[3](https://github.com/aiidateam/aiida-core/actions/runs/3129772900/jobs/5079265531#step:12:3)59e --label 817f70 --workdir /github/workspace --rm -e "SINGULARITY_ROOT" -e "pythonLocation" -e "LD_LIBRARY_PATH" -e "SLACK_WEBHOOK" -e "SLACK_ICON" -e "SLACK_CHANNEL" -e "SLACK_COLOR" -e "SLACK_TITLE" -e "SLACK_MESSAGE" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true --network github_network_bcded18fdafe[4](https://github.com/aiidateam/aiida-core/actions/runs/3129772900/jobs/5079265531#step:12:4)c0[6](https://github.com/aiidateam/aiida-core/actions/runs/3129772900/jobs/5079265531#step:12:6)[8](https://github.com/aiidateam/aiida-core/actions/runs/3129772900/jobs/5079265531#step:12:8)e03dac8b[9](https://github.com/aiidateam/aiida-core/actions/runs/3129772900/jobs/5079265531#step:12:9)b396fe -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/aiida-core/aiida-core":"/github/workspace" ghcr.io/rtcamp/action-slack-notify:v2.2.0
[ERROR] Secret `SLACK_WEBHOOK` is missing. Please add it to this action for proper execution.
Refer https://github.com/rtCamp/action-slack-notify for more information.
```

--- a/.github/workflows/nightly.yml
+++ b/.github/workflows/nightly.yml
@@ -76,7 +76,10 @@ jobs:
             run: .github/workflows/tests_nightly.sh
 
         -   name: Slack notification
-            if: always() && (steps.install.outcome == 'Failure' || steps.tests.outcome == 'Failure')
+            # Always run this step (otherwise it would be skipped if any of the previous steps fail) but only if the
+            # `install` or `tests` steps failed, and the `SLACK_WEBHOOK` is available. The latter is not the case for
+            # pull requests that come from forks. This is a limitation of secrets on GHA
+            if: always() && (steps.install.outcome == 'Failure' || steps.tests.outcome == 'Failure') && env.SLACK_WEBHOOK != null
             uses: rtCamp/action-slack-notify@v2
             env:
                 SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}

